schema/schema.go

```diff
@@ -187,8 +190,76 @@ func (s *Schema) ValidateData(data []byte) error {
 			return errors.Wrap(err, "failed to JSON remarshal data for validation")
 		}
 	}
+	if err = s.validate(schema.NewBytesLoader(data)); err != nil {
```

Comment:
Just a note (@klihub to confirm): I don't think that this validation is enabled by default, meaning that a producer will have to explicitly opt-in to validation here. Is there something that can be done when reading / writing a spec instead?

Reply:
@klihub Any comments on ^^ Thanks!

Reply:
Argh, sorry guys for not responding/missing this one...

Yes, that validation is not in effect/performed by default. You have to enable it by something like
```
import (
    "flag"
...
    "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
    "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi/validate"
    "github.com/container-orchestrated-devices/container-device-interface/schema"
...
)

var (
    schemaName string
)

// Set up JSON schema based Spec validation.
func setValidationSchema() error {
    s, err := schema.Load(schemaName)
    if err != nil {
        return fmt.Errorf("failed to load JSON schema %s: %w", schemaName, err)        
    }
    cdi.SetSpecValidator(validate.WithSchema(s))
    return nil
}

func main() {
    flag.StringVar(&schemaName, "schema", schema.BuiltinSchemaName,
        "validation schema file or URI, use " + schema.NoneSchemaName + "for no validation")
    ...
    flag.Parse()
    ...
    setValidationSchema()
    ...
}
```

There is a more accurate working example of that in `cmd/cdi/cmd/root.go` IIRC... 

The reason to not do this by default is to keey by default the extremely problematic `github.com/xeipuuv/gosjon*` packages out from the vendored dependencies of our downstream projects. This is good enough for most, but not all. For instance Kubernetes refuses to have those packages even in their `go.sum` (and therefore ATM importing this package).  Therefore we have #97 open...

Reply:
@zvonkok I have created https://github.com/zvonkok/container-device-interface/pull/1 which adds basic validation of annotations for devices and specs where other validation is currently performed. I think this circumvents the whole schema validation issue that we're looking at currently.

I believe that since we use `ReadSpec` to load the spec (which invokes `newSpec` where `spec.validate` is called), this should also validate the annotations when loading from disk.